hpx/hpx_init.hpp

```diff
@@ -193,7 +193,7 @@ namespace hpx
 #if !defined(HPX_HAVE_STATIC_LINKING)
     inline
 #endif
-    int init(init_params const& params = init_params());
+    int init(init_params& params);
```

Comment:
General comment on all these overloads. The last argument was optional because it gives us essentially double the overloads for free. Whatever the solution ends up being, we still want the following to work: `hpx::init()`, `hpx::init(argc, argv)`. The last argument can remain optional, but you won't be able to use `init_params()` as the default value like that anymore. It has to be a static again.

Reply:
Hmm, why do you want to pass the argument as a non-const reference? That would imply that the user supplied data will be changed by the function. Is that what we desire?

Reply:
Agreed, I've changed that as soon as I added the mutable for `startup` and `shutdown`, thanks !